File: src/main/java/main/users/allUsers.java
package main.users;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.Select;

public class allUsers {
    private WebDriver driver;
    public allUsers(WebDriver driver) {
        this.driver = driver;
    }
    private By administratorsTab = By.xpath("//img[@src='/images/info/large_tender_administrators.png']");
    private By tenderersTab = By.xpath("//img[@src='/images/info/large_tender_tenderers.png']");
    private By clientVendorsTab = By.xpath("//img[@src='/images/info/large_tender_company_users.png']");
    private By addUserButton = By.xpath("//option[contains(text(), 'to delete')]");
    private By removeUserButton = By.xpath("//input[@value='Delete']\n");
    private By okButton = By.xpath("//input[@value='OK']");
    private By selectPreviouslyInvitedUsersLink = By.xpath("//input[@value='OK']");
    public void clickAdmTab() {
        WebElement admTabElement = driver.findElement(administratorsTab);
        admTabElement.click();
    }
    public void clickTenderersTab() {
        WebElement tenderersTabElement = driver.findElement(tenderersTab);
        tenderersTabElement.click();
    }
    public void clickVendorsTab() {
        WebElement vendorsTabElement = driver.findElement(clientVendorsTab);
        vendorsTabElement.click();
    }


//    public void selectCategoryToDelete() {
//        Select checkBox = new Select(driver.findElement(deleteCategoryDropdown));
//        checkBox.selectByVisibleText("to delete");
//    }


    public void confirmOK() {
        WebElement okButtonElement = driver.findElement(okButton);
        okButtonElement.click();}

}

File: src/main/java/main/users/AllUsers.kt
package main.users

import org.openqa.selenium.By
import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement

open class AllUsers {
    private var driver: WebDriver? = null

    constructor(driver: WebDriver?) {
        this.driver = driver
    }

    private val administratorsTab: By = By.xpath("//img[@src='/images/info/large_tender_administrators.png']")
    private val tenderersTab: By = By.xpath("//img[@src='/images/info/large_tender_tenderers.png']")
    private val clientVendorsTab: By = By.xpath("//img[@src='/images/info/large_tender_company_users.png']")
    private val addUserButton: By = By.xpath("//img[@name='add_user']\n")
    private val removeUserButton: By = By.xpath("//img[@title='Remove user']\n")
    private val okButton: By = By.xpath("//input[@value='OK']")
    private val selectPreviouslyInvitedUsersLink: By = By.xpath("//input[@value='OK']")
    var emailField: By = By.xpath("//textarea[@name='emails']")
    private val closeWindowButton: By = By.xpath("//input[@value='Close window']")

    constructor()
    private fun textFieldForEmailInput(): WebElement {
        return driver!!.findElement(emailField)
    }

    fun clickAdmTab() {
        val admTabElement = driver!!.findElement(administratorsTab)
        admTabElement.click()
    }

    fun clickCloseWindowButton() {
        val closeWindowElement = driver!!.findElement(closeWindowButton)
        closeWindowElement.click()
    }

    fun clickTenderersTab() {
        val tenderersTabElement = driver!!.findElement(tenderersTab)
        tenderersTabElement.click()
    }

    fun clickVendorsTab() {
        val vendorsTabElement = driver!!.findElement(clientVendorsTab)
        vendorsTabElement.click()
    }

    fun clickAddUserButton() {
        val addUserButtonElement = driver!!.findElement(addUserButton)
        addUserButtonElement.click()
    }

    fun clickRemoveUserButton() {
        val removeUserButtonElement = driver!!.findElement(removeUserButton)
        removeUserButtonElement.click()
    }

    fun confirmOK() {
        val okButtonElement = driver!!.findElement(okButton)
        okButtonElement.click()
    }

    fun fillInTextFieldForEmailInput(emailField: String?): AllUsers {
        val textFieldForEmailInput = textFieldForEmailInput()
        textFieldForEmailInput.sendKeys(emailField)
        return this
    }

    fun enterEmail(email: String?) {
        val emailElement = driver!!.findElement(emailField)
        emailElement.clear()
        emailElement.sendKeys(email)
    }
}


